src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -1260,29 +1261,41 @@ internal bool TryInferConversion(IIsPatternOperation operation, out ConversionIn
                 }
 
                 // Handle is pattern operations with constant pattern.
-                if (operation is IIsPatternOperation patternOperation &&
-                    patternOperation.Pattern is IConstantPatternOperation constantPattern)
+                if (operation is IIsPatternOperation isPatternOperation)
                 {
-                    if (constantPattern.Value.ConstantValue.HasValue)
+                    IPatternOperation patternOperation = isPatternOperation.Pattern;
+                    bool direct = true;
+
+                    if (INegatedPatternOperationWrapper.IsInstance(patternOperation))
+                    {
+                        INegatedPatternOperationWrapper negatedPattern = INegatedPatternOperationWrapper.FromOperation(patternOperation);
+                        patternOperation = negatedPattern.Pattern;
```

Comment:
Can `.Children` work here? In this case we won't really need to introduce an operation wrapper

Reply:
It seems that `isPatternOperation.Pattern.Children.First()` matches `negatedPattern.Patten`
But is that making assumptions or can we rely on this?

Reply:
Since the pattern has only one child, accessing `.Children` looks good and is more simple. @mavasani @sharwell Are you okay with `.Children` approach instead of operation wrapper? The only thing we'll need in Lightup layer is a simple member in `OperationKindEx` enum.

Reply:
Are we not yet at a point where we can increment the Roslyn reference version?

Reply:
@sharwell If we can't upgrade, then https://github.com/dotnet/roslyn-analyzers/issues/5816 can help.

Reply:
Given we ship CA analyzers as part of .NET SDK, we can safely bump the Roslyn reference version to the one required by the last shipped SDK. @jmarolf do we know what is the minimum compiler/Roslyn version required by .NET6 SDK?

Reply:
@mavasani This can break people who get the analyzers via NuGet instead of the SDK?

I'm not opposed to that, but noting that upgrading might not be 100% safe.

Reply:
@Youssef1313 I think we want to explicitly call out (somewhere) that all versions of NetAnalyzers NuGet are not guaranteed to work on all versions of VS/compiler (basically Roslyn). It is reasonable to have a minimum required version of VS/Compiler associated with NetAnalyzers NuGet, which should match the version required by the corresponding .NET SDK. @jmarolf to confirm if this is a reasonable stance to take.

Reply:
@mavasani I agree with that.

Reply:
I must admit that I was surprised that the 6.0.0 version of the analyzer didn't demand 6.0 of the SDK or at least knew about all the new language features. The problem you have is that the cat is out of the bag, e.g. there might be people using the 6.0.0 of the analyzer with the 5.0 SDK. Changing it such that this doesn't work for the next version is a breaking change which would require a major version number increase to 7 which would confuse the heck out of developers.